pkg/controller/account/account_controller.go

```diff
@@ -81,44 +104,210 @@ type ReconcileAccount struct {
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+			reqLogger.Info("Failed to create user Access Key + ID")
+			return reconcile.Result{}, nil
+		}
+
+		// TODO: create secret details
+		userSecretInput := secretInput{
+			SecretName:              updatedAccount.Name,
+			NameSpace:               request.Namespace,
+			awsCredsUserName:        *userSecretInfo.AccessKey.UserName,
+			awsCredsSecretIDKey:     *userSecretInfo.AccessKey.AccessKeyId,
+			awsCredsSecretAccessKey: *userSecretInfo.AccessKey.SecretAccessKey,
+		}
+		userSecret := userSecretInput.newSecretforCR()
+		err = r.Client.Create(context.TODO(), userSecret)
+		if err != nil {
+			updatedAccount.Status.State = "Failed"
+			err = r.Client.Status().Update(context.TODO(), updatedAccount)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+			reqLogger.Info("Failed to create k8s user secret")
+			return reconcile.Result{}, nil
+		}
+
+		updatedAccount.Spec.IAMUserSecret = userSecret.ObjectMeta.Name
+		err = r.Client.Update(context.TODO(), updatedAccount)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		updatedAccount.Status.State = "Ready"
+		err = r.Client.Status().Update(context.TODO(), updatedAccount)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		reqLogger.Info("Account Ready to be claimed")
+
+		// set state to readys
+		// create ec2 instance , delete ec2 instance
+
+	}
 
 	return reconcile.Result{}, nil
 }
 
 // getAWSClient generates an awsclient
-func (r *ReconcileAccount) getAWSClient(awsAccessID, awsAccessSecret, region string) (awsclient.Client, error) {
+// function must include region
+// Pass in token if sessions requires a token
+// if it includes a secretName and nameSpace it will create credentials from that secret data
+// If it includes awsCredsSecretIDKey and awsCredsSecretAccessKey it will build credentials from those
+func (r *ReconcileAccount) getAWSClient(input newAwsClientInput) (awsclient.Client, error) {
+
+	// error is region is not included
+	if input.awsRegion == "" {
+		return nil, fmt.Errorf("getAWSClient:NoRegion: %v", input.awsRegion)
+	}
 
-	awsClient, err := r.awsClientBuilder(r.Client, awsAccessID, awsAccessSecret, region)
+	if input.secretName != "" && input.nameSpace != "" {
+		secret := &corev1.Secret{}
+		err := r.Client.Get(context.TODO(),
+			types.NamespacedName{
+				Name:      input.secretName,
+				Namespace: input.nameSpace,
+			},
+			secret)
+		if err != nil {
+			return nil, err
+		}
+		accessKeyID, ok := secret.Data[awsCredsSecretIDKey]
+		if !ok {
+			return nil, fmt.Errorf("AWS credentials secret %v did not contain key %v",
+				input.secretName, awsCredsSecretIDKey)
+		}
+		secretAccessKey, ok := secret.Data[awsCredsSecretAccessKey]
+		if !ok {
+			return nil, fmt.Errorf("AWS credentials secret %v did not contain key %v",
+				input.secretName, awsCredsSecretAccessKey)
+		}
+
+		fmt.Printf("STRINGS::: %s ::::: %s\n", accessKeyID, secretAccessKey)
```

Comment:
Probably can remove this for the time being

Reply:
I have the fix , turns out they are already base64 encoded. Fixing this now